### PR TITLE
Send Get Request To Server, Server Responds 'Hello'

### PR DESCRIPTION
**Pull request by DeKaylaPhillips**
# Send Get Request To Server, Server Responds 'Hello'

## Type of PR 

- [X] Refactor
- [ ] Feature
- [X] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This branch refactors the app's `listen()` method and tells the server to listen on Port 3000 instead of the initial Port 5000. The branch is also more modular in file structure, adds a test suite with a mock test to test the behavior of the server and its response, and includes changes to configuration files to resolve dependency and parsing errors. 

## Related Tickets & Documents

Screen Shot 2023-06-12 at 11 53 03 AM

- Closes <https://trello.com/c/NNpgqAq8/32-when-i-send-a-get-request-to-the-server-the-server-responds-with-hello>

## Added/updated tests?

- [X] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

### CODE REVIEW

**Comment by Naomi-Dennis** [comment]
Remove `package-lock.json` file

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 # Node 
 /node_modules
-package-lock.json
\ No newline at end of file
+
```
Comment (Naomi-Dennis) [inline .gitignore]: Remove whitespace

File: package.json [hunk 2]
```diff
@@ -4,19 +4,28 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
-    "dev": "ts-node ./src/server.ts",
-    "lint": "eslint src/**/*.ts"
+    "test": "jest",
+    "dev": "ts-node src/server.ts",
+    "lint": "eslint src/**/*.ts",
+    "lint tests": "eslint tests/**/*.spec.ts"
```
Comment (Naomi-Dennis) [inline package.json]: Good, but tests don't need linter checks!

File: tests/controller/index.spec.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+describe('Index Controller', () => {
+  test("responds 'Hello!'", () => {
+    const mockRequest: Partial<Request<any>> = {}
+    const mockResponse: Partial<Response<any>> = { send: jest.fn() }
+
+    indexController(mockRequest as Request, mockResponse as Response)
+
+    expect(mockResponse.send).toHaveBeenCalled()
+    expect(mockResponse.send).toHaveBeenCalledTimes(1)
```
Comment (Naomi-Dennis) [inline tests/controller/index.spec.ts]: Remove line 11

**Review by Naomi-Dennis — APPROVED** [review]
🚢

**Comment by Naomi-Dennis** [comment]
Clean up the commits. Try to get the commit count to either 8 or below. 
One way to do it is to use `git reset --soft HEAD~[# of commits]`

File: tests/controller/index.spec.ts (hunk 3 above)
Comment (DeKaylaPhillips) [reply tests/controller/index.spec.ts]: Line 11 has been removed. :)

File: package.json (hunk 2 above)
Comment (DeKaylaPhillips) [reply package.json]: Removed the lint script for the tests. :)

File: .gitignore (hunk 1 above)
Comment (DeKaylaPhillips) [reply .gitignore]: Removed the whitespace here.

**Comment by DeKaylaPhillips** [comment]
> Remove `package-lock.json` file

The package-lock is required by the CI and was added to GH to run properly. :)

**Review by Naomi-Dennis — APPROVED** [review]

**Pull request merged**